### PR TITLE
Remove iterator API. Remove TopN. Remove values API from PersistentMap

### PR DESCRIPTION
**Pull request by nearmax**

### CODE REVIEW

**Comment by nearmax** [comment]
> Why remove topN?

It is really really hard to implement since now our trie becomes unordered (key values are not ordered in any particular way, there is not iterator and there is no prefix search).
We can add it back later if there is an actual demand from one of our clients. We also would have to decide which implementation we would need to go with since there will be trade-offs.

**Comment by janedegtiareva** [comment]
The PR looks good. Some comments for release:
* we definitely want to communicate to devs that this change is happening. I would recommend to send out a comm before this goes live so that nobody is surprised by the change. 
* Is this going into the big backwards-incompatible release? Planning this out would be good to minimize chaos.

File: tests/test.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ passed=0
 
 testcase () {
 
-    res=`nearcore/target/debug/near-vm-runner-standalone --context-file=/tmp/context.json --config-file=/tmp/config.json --method-name=$1 --wasm-file=/tmp/main.wasm --input="{}"`
+    res=`/Users/zaver/Projects/nearcore/target/debug/near-vm-runner-standalone --context-file=/tmp/context.json --config-file=/tmp/config.json --method-name=$1 --wasm-file=/tmp/main.wasm --input="{}"`
```
Comment (vgrichina) [inline tests/test.sh]: leftover hardcoded path

**Comment by nearmax** [comment]
FYI @evgenykuzyakov I was implementing iterator-less collections before here, but then decided to keep iterators for the contracts and closed this PR.

**Pull request closed without merging**